### PR TITLE
Feat/UI touch ups

### PR DESCRIPTION
**Pull request by KiveshVirasamy**

### CODE REVIEW

File: src/app/components/trip/trip.component.html [hunk 1]
```diff
@@ -1,56 +1,51 @@
...
-                    <ul class="ml-4 space-y-2">
-                        <li class="text-gray-600">Description: {{item.description}}</li>
-                        <li class="text-gray-600">Start Date: {{item.startDate}}</li>
-                        <li class="text-gray-600">End Date: {{item.endDate}}</li>
-                        <li class="text-gray-600">Activities: <app-activity></app-activity></li>
+    <div class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-4">
+        <div *ngFor="let trip of trips">
+            <nz-card nzTitle={{trip.name}} class="bg-white border border-gray-200 shadow-md">
+                <ul class="space-y-2">
+                    <li><strong>Description:</strong> {{ trip.description }}</li>
```
Comment (MihlaliMgwebi) [inline src/app/components/trip/trip.component.html:6]: don't assume the trip and description exsit. Null coalesce (??) and display 'No description' for UI user feedback

Comment (MihlaliMgwebi) [reply src/app/components/trip/trip.component.html:6]: That applies to all of the dynamically retrieved data that can be null or undefined.

File: src/app/components/updateactivity/updateactivity.component.html [hunk 2]
```diff
@@ -1,6 +1,11 @@
+<nz-layout>
```
Comment (MihlaliMgwebi) [inline src/app/components/updateactivity/updateactivity.component.html:1]: maybe use router-outlet and separate these components?

File: src/app/shared/services/firestore.service.ts [hunk 3]
```diff
@@ -16,69 +15,90 @@ import { ITrips } from 'src/app/models/types/interfaces/trips';
...
+      .catch(() => {
+        window.alert('An error occurred while deleting the trip. Please try again later.');
+      });
   }
 
+
   getActivities() {
     const userId = localStorage.getItem('userId' ?? '');
-    const tripId = documentId();
+    const tripId = localStorage.getItem('tripId' ?? '');
```
Comment (MihlaliMgwebi) [inline src/app/shared/services/firestore.service.ts:65]: Nice null coalescing Kivesh!

**Review by MihlaliMgwebi — APPROVED** [review]
looks good. check my comments though then you may merge.

File: src/app/components/trip/trip.component.html (hunk 1 above)
Comment (KiveshVirasamy) [reply src/app/components/trip/trip.component.html:6]: Will add that

File: src/app/shared/services/firestore.service.ts (hunk 3 above)
Comment (KiveshVirasamy) [reply src/app/shared/services/firestore.service.ts:65]: Thanks Mish

**Pull request merged**